core/src/main/java/org/apache/accumulo/core/rpc/ThriftUtil.java

```diff
@@ -282,21 +283,39 @@ public static TTransport createClientTransport(HostAndPort address, int timeout,
         try {
           // Log in via UGI, ensures we have logged in with our KRB credentials
           final UserGroupInformation currentUser = UserGroupInformation.getCurrentUser();
+          final UserGroupInformation userForRpc;
+          if (AuthenticationMethod.PROXY == currentUser.getAuthenticationMethod()) {
+            // A "proxy" user is when the real (Kerberos) credentials are for a user
+            // other than the one we're acting as. When we make an RPC though, we need to make sure
+            // that the current user is the user that has some credentials.
+            if (currentUser.getRealUser() != null) {
+              userForRpc = currentUser.getRealUser();
+              log.trace("{} is a proxy user, using real user instead {}", currentUser, userForRpc);
+            } else {
+              // The current user has no credentials, let it fail naturally at the RPC layer (no ticket)
+              // We know this won't work, but we can't do anything else
+              log.warn("The current user is a proxy user but there is no underlying real user (RPCs will fail): {}", currentUser);
```

Comment:
Mostly, because it is a "best guess". This code is essentially reverting strict assertions because I thought I knew how things were going to work in practice.

I should update the comment to be a little less strongly worded.